### PR TITLE
Add `Finch.jl` package dev mode

### PR DESCRIPTION
**Pull request by mtsokol**
Hi @hameerabbasi @hameerabbasi,

To run `finch-tensor` test in `Finch.jl` to catch regressions we need a dev mode in `finch-tensor` (Also, I find it useful for local development when I want to test `finch-tensor` against unreleased Finch codebase).

Connected to https://github.com/willow-ahrens/Finch.jl/issues/539

### CODE REVIEW

File: src/finch/julia.py [hunk 1]
```diff
@@ -1,11 +1,20 @@
+import os
+
 import juliapkg
 
+_FINCH_NAME = "Finch"
 _FINCH_VERSION = "0.6.25"
 _FINCH_HASH = "9177782c-1635-4eb9-9bfb-d9dfa25e6bce"
+_FINCH_PKG_MODE = os.environ.get("FINCH_PKG_MODE", default="RELEASE")
```
Comment (hameerabbasi) [inline src/finch/julia.py]:
```suggestion
_FINCH_REPO_PATH = os.environ.get("FINCH_REPO_PATH", default=None)
```

File: src/finch/julia.py [hunk 2]
```diff
@@ -1,11 +1,20 @@
...
 
-deps = juliapkg.deps.load_cur_deps()
-if deps.get("packages", {}).get("Finch", {}).get("version", None) != _FINCH_VERSION:
-    juliapkg.add("Finch", _FINCH_HASH, version=_FINCH_VERSION)
+if _FINCH_PKG_MODE == "RELEASE":
```
Comment (hameerabbasi) [inline src/finch/julia.py]:
```suggestion
if _FINCH_REPO_PATH:  # Also account for empty string
```

File: src/finch/julia.py [hunk 3]
```diff
@@ -1,11 +1,20 @@
...
+    deps = juliapkg.deps.load_cur_deps()
+    if deps.get("packages", {}).get(_FINCH_NAME, {}).get("version", None) != _FINCH_VERSION:
+        juliapkg.add(_FINCH_NAME, _FINCH_HASH, version=_FINCH_VERSION)
+elif _FINCH_PKG_MODE == "DEV":
+    juliapkg.add(_FINCH_NAME, _FINCH_HASH, path=os.environ["FINCH_REPO_PATH"], dev=True)
```
Comment (hameerabbasi) [inline src/finch/julia.py]:
```suggestion
    juliapkg.add(_FINCH_NAME, _FINCH_HASH, path=_FINCH_REPO_PATH, dev=True)
```

**Review by hameerabbasi — CHANGES_REQUESTED** [review]
How about doing things via one environment var instead of two to make things easier for the user?

File: src/finch/julia.py [hunk 4]
```diff
@@ -1,11 +1,18 @@
...
 
+_FINCH_NAME = "Finch"
 _FINCH_VERSION = "0.6.25"
 _FINCH_HASH = "9177782c-1635-4eb9-9bfb-d9dfa25e6bce"
+_FINCH_REPO_PATH = os.environ.get("FINCH_REPO_PATH", default=None)
 
-deps = juliapkg.deps.load_cur_deps()
-if deps.get("packages", {}).get("Finch", {}).get("version", None) != _FINCH_VERSION:
-    juliapkg.add("Finch", _FINCH_HASH, version=_FINCH_VERSION)
+if _FINCH_REPO_PATH:  # Also account for empty string
```
Comment (hameerabbasi) [inline src/finch/julia.py:10]:
```suggestion
if not _FINCH_REPO_PATH:  # Also account for empty string
```

My bad there... 😅

Comment (mtsokol) [reply src/finch/julia.py:10]: It's already correct! I use dev mode if it's true.

**Review by hameerabbasi — APPROVED** [review]

**Pull request merged**